### PR TITLE
properly sort out TCNP subnets in CF stack templates

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

**Comment by MarcelMue** [comment]
PTAL - this should be correct now except for the missing CIDRs

File: service/controller/clusterapi/v29/resource/tcnp/create.go [hunk 1]
```diff
@@ -288,22 +292,23 @@ func (r *Resource) newSubnets(ctx context.Context, cl v1alpha1.Cluster, md v1alp
...
-				NameSuffix: strings.ToUpper(a.Name),
+				Name: key.SanitizeCFResourceName(key.PrivateSubnetRouteTableAssociationName(a.Name)),
 			},
 			TCCP: template.ParamsMainSubnetsListItemTCCP{
 				Subnet: template.ParamsMainSubnetsListItemTCCPSubnet{
-					ID: "", // TODO
+					ID: key.SanitizeCFResourceName(key.PublicSubnetName(a.Name)),
 					RouteTable: template.ParamsMainSubnetsListItemTCCPSubnetRouteTable{
-						ID: "", // TODO
+						ID: key.SanitizeCFResourceName(key.PublicRouteTableName(a.Name)),
```
Comment (xh3b4sd) [inline service/controller/clusterapi/v29/resource/tcnp/create.go]: Let's call these `Name` as well please.

Comment (MarcelMue) [reply service/controller/clusterapi/v29/resource/tcnp/create.go]: Done :heavy_check_mark:

**Comment by xh3b4sd** [comment]
I think we can merge if you like and fix the private subnet CIDR later.

**Pull request merged**